src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    switch (action.type) {
```

Comment:
Better destructure all the needed props of action (type, extraData, keysToRemove)